### DISCUSSION TITLE
Debugging problem

### DISCUSSION THREAD

**Discussion by rvwilliams88** [unoplatform/uno#17072] [General]
I have changed the PeopleApp PeopleService.cs to pick up data from an external database.  However when I run the code it just show "An error occurred" and nothing else.  The service I am using is in current use in several other non-Uno apps but obviously something in it disagrees with Uno.  Is it possible that the Uno Platform is catching an error and changing the message to An error occurred?  As this message isn't produced by my service, how best should I identify the source?

```csharp
namespace PeopleApp;

public partial record Person(string? FirstName, string? LastName);

public interface IPeopleService
{
    ValueTask<IImmutableList<Person>> GetPeopleAsync(CancellationToken ct);
}

public class PeopleService : IPeopleService
{
    AuthorService? authorService { get; set; }
    public async ValueTask<IImmutableList<Person>> GetPeopleAsync(CancellationToken ct)
    {
        authorService = new AuthorService();
        var people = new Person[6];
        List<Author>? authors = new List<Author>();
        authors = await authorService!.getAuthors();
        foreach (Author? author in authors?.Take(6)!)
        {
            people.Append(new Person(FirstName: author.first, LastName: author.last));
        }
        return people.ToImmutableList();
    }
}
```

**Comment by carldebilly** [comment]
Where do you see that error? On which platform?

BTW, I would suggest improving your code like this...

```csharp
public class PeopleService : IPeopleService
{
    private readonly AuthorService _authorService;

    public PeopleService(AuthorService? authorService = null)
    {
        _authorService = authorService ?? new AuthorService();
    }

    public async ValueTask<IImmutableList<Person>> GetPeopleAsync(CancellationToken ct)
    {
        if (await _authorService.getAuthors() is null or { Length: 0 } authors)
        {
            return ImmutableList<Person>.Empty;
        }

        var people = authors.Take(6).Select(a=>a.AsPerson()).ToImmutableList();
        return people;
    }
}
```

**Comment by rvwilliams88** [comment]
Sorry, your code doesn't compile. (no definition for 'authors' , Author doesn't contain AsPerson, authors is a List so no Length)

What I've done is taken the standard PeopleApp sample which is XAML and C# with MVUX and converted the XAML to C# markup.  That works fine with the supplied PeopleService where the value of 'people' is built into the code.  Now I want to replace that with values read from a database.  The code for the database is in a separate code library that is used in several other C# apps.

To try to identify where the error is occurring, I've put a 'throw new Exception' right at the start of the modified PeopleService and my message isn't displayed and this suggests to me that the error is being generated elsewhere, but where exactly?

Regards

Richard.
****************************************************
Richard Williams
email: ***@***.******@***.***>
mobile: 07981 734697
****************************************************

**Comment by rvwilliams88** [comment]
SOLVED.  The problem was not related to the Uno Platform.

**Comment by rvwilliams88** [comment]
Now resolved. Problem was not related to the Uno Platform.

Regards

Richard.
****************************************************
Richard Williams
email: ***@***.******@***.***>
mobile: 07981 734697
****************************************************
